client/components/UI/DateTime.tsx

```diff
@@ -12,37 +12,82 @@ import './style.scss';
  * @name DateTime
  * @description DateTime component to display text formatted date and time
  */
-function DateTime({date, withTime, withDate, withYear, padLeft, toBeConfirmed}) {
+function DateTime({
+    date,
+    withTime,
+    withDate,
+    withYear,
+    padLeft,
+    toBeConfirmed,
+    allDay,
+    isEndEventDateTime,
+    noEndTime,
+    setHideDash,
+    multiDay,
+}) {
     const {gettext} = superdeskApi.localization;
     const dateFormat = appConfig.planning.dateformat;
     const timeFormat = appConfig.planning.timeformat;
-    const newTimeFormat = toBeConfirmed ? `[${gettext(`Time ${gettext('TBC')}`)}]` : timeFormat;
-    let format = withYear ? dateFormat : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
+    const newTimeFormat = toBeConfirmed
+        ? `[${gettext(`Time ${gettext('TBC')}`)}]`
+        : timeFormat;
+    let format = withYear
+        ? dateFormat
+        : dateFormat.replace(/([\/\#.-]y+)|(y+[\/\#.-])/gi, '');
     let dateTimeFormat = [
         withDate ? format : null,
         withTime ? newTimeFormat : null,
-    ].filter((d) => d).join('\u00a0'); // &nbsp;
+    ]
+        .filter((d) => d)
+        .join('\u00a0'); // &nbsp;
+
+    let eventEndDate;
+
+    if (noEndTime && !multiDay) {
+        eventEndDate = null;
+        setHideDash && setHideDash(true);
+    } else if (allDay && multiDay) {
+        eventEndDate = moment(date).format(dateTimeFormat)
+            .slice(0, 5) || '';
+    } else if (noEndTime && multiDay) {
+        eventEndDate = moment(date).format(dateTimeFormat)
+            .slice(0, 5) || '';
+    } else if (allDay && !multiDay) {
+        setHideDash && setHideDash(true);
```

Comment:
I don't see the need for this setHideDash call, doesn't the parent component have all the info?

Reply:
We have to use setHideDash because parent component doesn't have that info on which we have to hide dash. We are hiding dash when we don't want to show end date and the whole logic resides in child component. That's why i thought it would be better to use this here.

Reply:
given that the child component gets all the info from the parent it makes me think that the parent has all the info

Reply:
For the start and finish dates, we do, however, use a child component twice in the parent component, with a dash in between. Date logic is included in the child component, and since the end date is hidden from the logic of the child component, we must use it.